src/Workspaces/Remote/Core/AbstractAssetProvider.cs

```diff
@@ -40,11 +46,17 @@ public async Task<SolutionInfo> CreateSolutionInfoAsync(Checksum solutionChecksu
         await this.GetAssetHelper<ProjectStateChecksums>().GetAssetsAsync(
             AssetPathKind.ProjectStateChecksums,
             solutionChecksums.Projects.Checksums,
-            static (_, projectStateChecksums, tuple) => tuple.projectsTasks.Add(tuple.@this.CreateProjectInfoAsync(projectStateChecksums, tuple.cancellationToken)),
-            (@this: this, projectsTasks, cancellationToken),
+            static (_, projectStateChecksums, args) =>
+            {
+                var (@this, projectsTasks, assemblyLoaderProvider, cancellationToken) = args;
+                projectsTasks.Add(@this.CreateProjectInfoAsync(projectStateChecksums, assemblyLoaderProvider, cancellationToken));
+            },
+            (@this: this, projectsTasks, assemblyLoaderProvider, cancellationToken),
             cancellationToken).ConfigureAwait(false);
 
-        var analyzerReferences = await this.GetAssetsArrayAsync<AnalyzerReference>(AssetPathKind.SolutionAnalyzerReferences, solutionChecksums.AnalyzerReferences, cancellationToken).ConfigureAwait(false);
+        var isolatedAnalyzerReferences = await this.CreateIsolatedAnalyzerReferencesAsync(
```

Comment:
a critical thing here is taht all the OOP calls to fetch analyzer references should go through this helper.  This is enforced by having the normal `.GetAssetsArrayAsync<T>` call throw if `T` is a `AnalyzerReference`.

Reply:
there are 3 places in OOP where we hydrate this list of references, and all were updated to call the new helper.